agent/xds/server.go

```diff
@@ -90,6 +95,18 @@ type Server struct {
 	CfgMgr       ConfigManager
 	Authz        ConnectAuthz
 	ResolveToken ACLResolverFunc
+	// AuthCheckFrequency is how often we should re-check the credentials used
+	// during a long-lived gRPC Stream after it has been initially established.
+	// This is only used during idle periods of stream interactions (i.e. when
+	// there has been no recent DiscoveryRequest).
+	AuthCheckFrequency time.Duration
+}
+
+// Initialize will finish configuring the Server for first use.
+func (s *Server) Initialize() {
+	if s.AuthCheckFrequency == 0 {
+		s.AuthCheckFrequency = DefaultAuthCheckFrequency
```

Comment:
Why not just use the `acl.token_ttl` configuration for this? I think the default for it is a much lower value (higher frequency of checks) but is that too much of a problem?

Reply:
Mostly because I did not know that there was a `token_ttl` config setting. 

If the token is already going to be cached for that long, and assuming this existing xDS authz checking function call I'm repurposing goes through the cache correctly, then it's probably fine to switch it.

It will be as if all envoy instances were doing two RPCs a minute requiring authz checks.

Reply:
The only reason I could think of was that maybe that would cause too many RPCs. By default we have a 30s token ttl. So it would be 10 times more RPCs than the way you coded it up. Its probably not an issue and it would allow the value to be tuned via the config.

Reply:
Are we talking about re-checking auth every 30 seconds for every active envoy instance? That seems kinda expensive to me in terms of RPCs.

Consider that in general most clients are doing 10min blocking queries that also have the same semantic that if the token is revoked during the query they will still be delivered (possibly filtered) results at the end.

In my mind the value of active re-auth checking is as an optimisation to early fail an idle Envoy when it's token is revoked rather than wait for it to need  new result to be sent and check then. As such waking up and performing an RPC to do that every 30 seconds seems like overkill to me....

I'd argue for 5 or 10 mins being sufficient 0 it's still better than blocking query clients.